import-service/handlers/importProductsFile.ts

```
@@ -0,0 +1,41 @@
+import {APIGatewayProxyEvent, APIGatewayProxyHandler} from 'aws-lambda';
+import 'source-map-support/register';
+import * as AWS from 'aws-sdk';
+
+import {AWS_REGION, BUCKET_NAME, SOURCE_DIR} from '../constants';
+import {RESPONSE} from '../../shared/constants/responses';
+
+export const importProductsFile = async (event: APIGatewayProxyEvent) => {
+  const s3 = new AWS.S3({region: AWS_REGION});
+
+  const {name} = event.queryStringParameters;
+
+  if (!name) {
```

Also it would be nice to validate the file type here as the parsing function is working only with .csv files.

---

text/csv types are allowed only on bucket level (line 24)

if type does not match then bucket itself will return 404 forbidden.

But I agree that it would be better to return a meaningful error message at this stage.
Thank you